src/ShaderGen/ShaderMethodVisitor.cs

```diff
@@ -211,6 +211,19 @@ public override string VisitInvocationExpression(InvocationExpressionSyntax node
                     ValidateBuiltInMethod(method);
                 }
 
+                if (type == "ShaderGen.ShaderBuiltins" && method == "SampleComparisonLevelZero")
```

Comment:
This condition is partially redundant with the comparison above. I wonder if we should make "ValidateBuiltInMethod" more general ("ProcessBuiltInMethod") and add this logic there.?